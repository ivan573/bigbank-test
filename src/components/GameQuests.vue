<template>
  <ul class="game-quests">
    <li v-for="quest in quests" class="game-quests__quest" :key="quest.adId" @click="handleQuestClick(quest.adId)">
      <p>{{ quest.message }}</p>
      <p>Reward: {{ quest.reward }}</p>
      <p>Difficulty: {{ quest.probability }}</p>
      <p>Expires in {{ quest.expiresIn }} turns</p>
    </li>
  </ul>
</template>

<script>
export default {
  name: 'GameQuests',
  props: {
    quests: Array
  },
  methods: {
    handleQuestClick(id) {
      this.$emit('questClick', id)
    }
  }
}
</script>

<style lang="scss" scoped>
  $gap: 16px;

  .game-quests {
    padding: 0;
    display: flex;
    flex-wrap: wrap;
    gap: $gap;
    list-style: none;

    &__quest {
      padding: 8px;
      width: calc((100% - $gap) / 2);
      border: 1px solid black;
      border-radius: 8px;
      box-sizing: border-box;
      cursor: pointer;
      background-color: white;
      transition: 0.1s;

      &:hover {
        transform: scale(1.2);
      }
    }
  }
</style>
