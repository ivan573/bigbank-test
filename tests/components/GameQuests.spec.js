import { shallowMount } from '@vue/test-utils'
import GameQuests from '@/components/GameQuests.vue'

describe('GameQuests.vue', () => {
  const quests = [
    {
      adId: 'Ln859bV7',
      expiresIn: 4,
      message: 'Help Menes Rennold to transport a magic turnips to grassland in Everdrone',
      probability: 'Piece of cake',
      reward: 15

    },
    {
      adId: 'lBxlpzxe',
      expiresIn: 5,
      message: 'Help Viriato Ásmundsdóttir to sell an unordinary water on the local market',
      probability: 'Walk in the park',
      reward: 10

    },
    {
      adId: 'bauDDuqQ',
      expiresIn: 3,
      message: 'Help Chingis Chandler to fix their horse',
      probability: 'Quite likely',
      reward: 8

    }
  ]

  const wrapper = shallowMount(GameQuests, {
    propsData: { quests }
  })

  const questElements = wrapper.findAll('.game-quests__quest')

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('displays all of the items', () => {
    expect(questElements.length).toBe(quests.length)
  })

  it('displays all item data correctly', () => {
    questElements.wrappers.forEach((element, index) => {
      const itemDataElements = element.findAll('.game-quests__quest-text').wrappers
      expect(itemDataElements[0].text()).toBe(quests[index].message)
      expect(itemDataElements[1].text()).toBe(`Reward: ${quests[index].reward}`)
      expect(itemDataElements[2].text()).toBe(`Difficulty: ${quests[index].probability}`)
      expect(itemDataElements[3].text()).toBe(`Turns before expiration: ${quests[index].expiresIn}`)
    })
  })

  it('emits itemClick event with the item id on item click', () => {
    questElements.wrappers.forEach((element, index) => {
      element.trigger('click')
      expect(wrapper.emitted().questClick[index]).toEqual([quests[index].adId])
    })
  })
})
