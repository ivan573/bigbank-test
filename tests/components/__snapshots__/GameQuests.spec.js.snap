// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameQuests.vue matches snapshot 1`] = `
<ul
  class="game-quests"
>
  <li
    class="game-quests__quest"
  >
    <p
      class="game-quests__quest-text"
    >
      Help Menes Rennold to transport a magic turnips to grassland in Everdrone
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Reward: 15
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Difficulty: Piece of cake
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Turns before expiration: 4
    </p>
  </li>
  <li
    class="game-quests__quest"
  >
    <p
      class="game-quests__quest-text"
    >
      Help Viriato Ásmundsdóttir to sell an unordinary water on the local market
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Reward: 10
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Difficulty: Walk in the park
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Turns before expiration: 5
    </p>
  </li>
  <li
    class="game-quests__quest"
  >
    <p
      class="game-quests__quest-text"
    >
      Help Chingis Chandler to fix their horse
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Reward: 8
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Difficulty: Quite likely
    </p>
     
    <p
      class="game-quests__quest-text"
    >
      Turns before expiration: 3
    </p>
  </li>
</ul>
`;
